docker-compose.yml

```diff
@@ -0,0 +1,61 @@
+version: '3.7'
+
+services:
+  mysql:
+    image: mysql:8.0
+    command: --default-authentication-plugin=mysql_native_password
+    environment:
+      MYSQL_ROOT_PASSWORD: password
+    volumes:
+      - mysql:/var/lib/mysql
+
+  postgres:
+    image: postgres:13.2
+    environment:
+      POSTGRES_USER: root
+      POSTGRES_PASSWORD: password
+    volumes:
+      - postgres:/var/lib/postgresql/data
+
+  app:
+    build:
+      context: .dockerdev
+      dockerfile: Dockerfile
+      args:
+        RUBY_VERSION: "2.7.2"
+        PG_VERSION: 13
+        NODE_VERSION: 14
+        MYSQL_VERSION: "8.0"
+        BUNDLER_VERSION: 2
+    image: solidus-3.0.0.rc1
+    command: bash -c "(bundle check || bundle) && tail -f Gemfile"
+    environment:
+      CAPYBARA_DRIVER: selenium_chrome_headless_docker_friendly
+      DB_USERNAME: root
+      DB_PASSWORD: password
+      RAILS_VERSION: ${RAILS_VERSION:-~> 6.1.0}
+      DB_ALL: "1"
+      DB_MYSQL_HOST: mysql
+      DB_POSTGRES_HOST: postgres
+      HISTFILE: "/home/solidus_user/history/bash_history"
+      MYSQL_HISTFILE: "/home/solidus_user/history/mysql_history"
+    ports:
+      - "3000:3000"
+    volumes:
+      - .:/home/solidus_user/app:cached
+      - bundle:/home/solidus_user/gems
+      - history:/home/solidus_user/history
+      - .dockerdev/.psqlrc:/home/solidus_user/.psqlrc
+    tty: true
+    stdin_open: true
+    tmpfs:
+      - /tmp
+    depends_on:
+      - mysql
+      - postgres
```

Comment:
You could also evaluate adding `user` to set the uid/gid (but this must be tested carefully because it can be tricky to handle all the permissions accordingly): `user: "${UID}:${GID}"`

Reply:
Sorry, but I don't understand this comment :sweat_smile: Where should I add the user?

Reply:
hey @blocknotes any extra info for this one? :slightly_smiling_face:

Reply:
It's a minor point @waiting-for-dev - but if you are curious I mean with just a `user` key.

In my setup I have a common block like:
```yml
x-defaults:
  app: &default_app
    user: "${UID}:${GID}"
```
And then:
```yml
services:
  app:
    <<: *default_app
```
In my case I also have a .env with:
```
UID=${UID}
GID=${GID}
```

Reply:
Ok, I didn't know it was allowed to provide the user through docker-compose. However, we're performing some actions in the `Dockerfile` for a `solidus_user` user. If a developer could provide a different user through `docker-compose.yml` it would be inconsistent. If I'm not wrong, to enforce this consistency we should add a `--build-arg` and the user should be aware to make it match with what it's configured in `docker-compose.yml`. It would complicate things. Please, if you know a solution for this let me know. User management in containers is something that keeps bugging me.

Reply:
Yeah, probably a `--build-arg` (or perhaps an exported env var) would be necessary to set a specific user. It seems tricky in this case to use the `user` parameter.